File: macos/Scanner/Sources/Scanner/Detectors/CursorDetector.swift
public struct CursorDetector: Detector {
    public let id = "cursor"
    public let displayName = "Cursor"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        let mcpPath = homePath(fs, ".cursor/mcp.json")
        let settingsPath = homePath(fs, "Library/Application Support/Cursor/User/settings.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: ["/Applications/Cursor.app", homePath(fs, "Applications/Cursor.app")]
        )

        if let text = try readIfPresent(fs, path: mcpPath) {
            facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: mcpPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: mcpPath)
        }

        if let text = try readIfPresent(fs, path: settingsPath) {
            facts.settings.append(try FactParsers.parseCursorPrivacyMode(text, appId: id, configPath: settingsPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: settingsPath)
        }

        return facts
    }
}

File: windows/Scanner/Detectors/CursorDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class CursorDetector : IDetector
{
    public string Id => "cursor";
    public string DisplayName => "Cursor";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var mcpPath = DetectorSupport.HomePath(fs, ".cursor/mcp.json");
        var settingsPath = Path.Combine(fs.ApplicationDataDirectory, "Cursor", "User", "settings.json");
        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [Path.Combine(fs.LocalApplicationDataDirectory, "Programs", "cursor")]
        );

        var mcpText = DetectorSupport.ReadIfPresent(fs, mcpPath);
        if (mcpText is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(mcpText, Id, mcpPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, mcpPath);
        }

        var settingsText = DetectorSupport.ReadIfPresent(fs, settingsPath);
        if (settingsText is not null)
        {
            facts.Settings.Add(FactParsers.ParseCursorPrivacyMode(settingsText, Id, settingsPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, settingsPath);
        }

        return facts;
    }
}
